File: app/src/main/java/com/game/airplane/GameView.java
package com.game.airplane;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.util.Log;
import android.view.DragEvent;
import android.view.MotionEvent;
import android.view.SurfaceView;

import com.game.airplane.components.Background;
import com.game.airplane.components.Flight;

public class GameView extends SurfaceView implements Runnable {

    private Thread thread;
    private boolean isPlaying;
    private int screenX, screenY;
    public static float screenRatioX, screenRatioY;
    private Paint paint;
    private Flight flight;
    private Background background1, background2;

    public GameView(Context context, int screenX, int screenY) {
        super(context);
        this.screenX = screenX;
        this.screenY = screenY;

        screenRatioX = 1920f / screenX;
        screenRatioY = 1080f / screenY;

        background1 = new Background(screenX, screenY, getResources());
        background2 = new Background(screenX, screenY, getResources());

        flight = new Flight(screenY, getResources());

        background2.x = screenX;
        paint = new Paint();
    }

    @Override
    public void run() {
        while (isPlaying) {
            update();
            draw();
            sleep();
        }
    }

    private void update() {
        background1.x -= 12;
        background2.x -= 12;

        if (background1.x + background1.background.getWidth() < 0) {
            background1.x = screenX;
        }

        if (background2.x + background2.background.getWidth() < 0) {
            background2.x = screenX;
        }

        //nao deixa passar do limite de cima da tela
        if (flight.y < 0) {
            flight.y = 0;
        }

        //não deixa passar do limite de baixo da tela
        if (flight.y >= screenY - flight.height) {
            flight.y = screenY - flight.height;
        }

    }

    private void draw() {
        if (getHolder().getSurface().isValid()) {
            Canvas canvas = getHolder().lockCanvas();

            canvas.drawBitmap(background1.background, background1.x, background1.y, paint);
            canvas.drawBitmap(background2.background, background2.x, background2.y, paint);

            canvas.drawBitmap(flight.getFlight(), flight.x, flight.y, paint);

            getHolder().unlockCanvasAndPost(canvas);
        }
    }

    private void sleep() {
        try {
            Thread.sleep(17);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
    }

    public void resume() {
        isPlaying = true;
        thread = new Thread(this);
        thread.start();
    }

    public void pause() {
        try {
            isPlaying = false;
            thread.join();
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        flight.y = (int) event.getY();
        return true;
    }
}

File: app/src/main/java/com/game/airplane/GameView.kt
package com.game.airplane

import android.content.Context
import android.graphics.Paint
import android.view.MotionEvent
import android.view.SurfaceView
import com.game.airplane.components.Background
import com.game.airplane.components.Flight

class GameView(context: Context?, private val screenX: Int, private val screenY: Int) :
    SurfaceView(context), Runnable {
    private var thread: Thread? = null
    private var isPlaying = false
    private val paint: Paint
    private val flight: Flight
    private val background1: Background
    private val background2: Background

    init {
        screenRatioX = 1920f / screenX
        screenRatioY = 1080f / screenY
        background1 = Background(screenX, screenY, resources)
        background2 = Background(screenX, screenY, resources)
        flight = Flight(screenY, resources)
        background2.x = screenX
        paint = Paint()
    }

    override fun run() {
        while (isPlaying) {
            update()
            draw()
            sleep()
        }
    }

    private fun update() {
        background1.x -= 11
        background2.x -= 11
        if (background1.x + background1.background.width < 0) {
            background1.x = screenX
        }
        if (background2.x + background2.background.width < 0) {
            background2.x = screenX
        }

        //nao deixa passar do limite de cima da tela
        if (flight.y < 0) {
            flight.y = 0
        }

        //não deixa passar do limite de baixo da tela
        if (flight.y >= screenY - flight.height) {
            flight.y = screenY - flight.height
        }
    }

    private fun draw() {
        if (holder.surface.isValid) {
            val canvas = holder.lockCanvas()
            canvas.drawBitmap(
                background1.background,
                background1.x.toFloat(),
                background1.y.toFloat(),
                paint
            )
            canvas.drawBitmap(
                background2.background,
                background2.x.toFloat(),
                background2.y.toFloat(),
                paint
            )
            canvas.drawBitmap(flight.flight, flight.x.toFloat(), flight.y.toFloat(), paint)
            holder.unlockCanvasAndPost(canvas)
        }
    }

    private fun sleep() {
        try {
            Thread.sleep(17)
        } catch (e: InterruptedException) {
            throw RuntimeException(e)
        }
    }

    fun resume() {
        isPlaying = true
        thread = Thread(this)
        thread!!.start()
    }

    fun pause() {
        try {
            isPlaying = false
            thread!!.join()
        } catch (e: InterruptedException) {
            throw RuntimeException(e)
        }
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        if (event.x < 400) {
            flight.y = event.y.toInt()
            flight.y -= 60
            if (flight.y >= screenY - flight.height) {
                flight.y = screenY - flight.height
            }
            if (flight.y < 0) {
                flight.y = 0
            }
        }
        return true
    }

    companion object {
        var screenRatioX: Float = 0.0f
        var screenRatioY: Float = 0.0f
    }
}